cpp-inference/inference_scaling_imagenet.cpp

```diff
@@ -74,64 +74,117 @@ void run_mnist(const std::string& model_name,
     std::cout<<"Connecting clients"<<std::endl<<std::flush;
 
   double constructor_start = MPI_Wtime();
   bool cluster = get_cluster_flag();
   bool is_colocated = get_colo();
-  SmartRedis::Client client(cluster);
+  SmartRedis::Client client(cluster, context);
   double constructor_end = MPI_Wtime();
   double delta_t = constructor_end - constructor_start;
   timing_file << rank << "," << "client()" << ","
               << delta_t << std::endl << std::flush;
+  //print rank # storing client() for debugging
+  std::string client_text = "client() time stored for rank: ";
+  client_text += std::to_string(rank);
+  log_data(context, LLDebug, client_text);
+
   std::string device = get_device();
   int num_devices = get_num_devices();
   bool use_multigpu = (0 == device.compare("GPU")) && num_devices > 1;
   bool should_set = get_set_flag();
   std::string model_key = "resnet_model";
   std::string script_key = "resnet_script";
+  log_data(context, LLDebug, "Running rank: " + std::to_string(rank) + "with vars should_set: " + std::to_string(should_set) + " - num_device: " + std::to_string(num_devices) + " - use_multigpu: " + std::to_string(use_multigpu) + " - is_coloated: " + std::to_string(is_colocated) + " - cluster: " + std::to_string(cluster));
   if (should_set) {
     int batch_size = get_batch_size();
     int n_clients = get_client_count();
+    log_data(context, LLDebug, "Running rank: " + std::to_string(rank) + "with batch_size: " + std::to_string(batch_size) + " and n_clients: " + std::to_string(n_clients));
     if (!is_colocated && rank == 0) {
+      log_data(context, LLDebug, "Rank: " + std::to_string(rank) + " entered non-colocated if block");
+
       std::cout<<"Setting Resnet Model from scaling app" << std::endl;
+      log_data(context, LLInfo, "Setting Resnet Model from scaling app");
+
       std::cout<<"Setting with batch_size: " << std::to_string(batch_size) << std::endl;
+      std::string batch_text = "Setting with batch_size: ";
+      batch_text += std::to_string(batch_size);
+      log_data(context, LLInfo, batch_text);
+
       std::cout<<"Setting on device: " << device << std::endl;
+      std::string device_text = "Setting on device: ";
+      device_text += device;
+      log_data(context, LLInfo, device_text);
+
       std::cout<<"Setting on " << std::to_string(num_devices) << " devices" <<std::endl << std::flush;
+      std::string set_text = "Setting on ";
+      set_text += std::to_string(num_devices);
+      set_text += " devices";
+      log_data(context, LLInfo, set_text);
+
       std::string model_filename = "./resnet50." + device + ".pt";
 
       if (use_multigpu) {
         client.set_model_from_file_multigpu(model_key, model_filename, "TORCH", 0, num_devices, batch_size);
+        log_data(context, LLDebug, "Use_multigpu - model_key: " + model_key + " model_filename: " + model_filename + " num_devices: " + std::to_string(num_devices) + " batch_size: " + std::to_string(batch_size));
```

Comment:
split and check log message length

Reply:
what do u use to check your line length? Do u have a package u downloaded?